lib/core/uploaderError.js

```diff
@@ -17,29 +17,43 @@
 
 var sundial = require('sundial');
 
-// Create a new object, that prototypally inherits from the Error constructor.
-function UploaderError(message, stage, error) {
-
-  this.name = this.constructor.name;
-  this.message = message;
+// if another error is passed in then attach it
```

Comment:
I don't want to loose any history that may be attached to the original message we caught - this is a simple way to do that.